File: bidly-fastapi/app/models/medio_pago.py
from sqlalchemy import Column, Integer, String, Numeric, ForeignKey
from app.database import Base


class MedioPago(Base):
    __tablename__ = "mediosdepago"

    identificador = Column(Integer, primary_key=True, autoincrement=True)
    cliente       = Column(Integer, ForeignKey("clientes.identificador"))
    tipo          = Column(String)
    numerotarjeta = Column(String)
    vencimiento   = Column(String)
    titular       = Column(String)
    numerocuenta  = Column(String)
    banco         = Column(String)
    numerocheque  = Column(String)
    montocheque   = Column(Numeric(precision=12, scale=2))
    verificado    = Column(String, default="no")

File: bidly-backend/src/main/java/com/bidly/bidly_backend/model/MedioPago.java
package com.bidly.bidly_backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "mediosdepago")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MedioPago {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long identificador;

    @ManyToOne
    @JoinColumn(name = "cliente")
    private Cliente cliente;

    private String tipo;
    @Column(name = "numerotarjeta")
    private String numeroTarjeta;
    private String vencimiento;
    private String titular;

    @Column(name = "numerocuenta")
    private String numeroCuenta;
    private String banco;

    @Column(name = "numerocheque")
    private String numeroCheque;

    @Column(name = "montocheque")
    private BigDecimal montoCheque;
    private String verificado;
}
